### PR TITLE
Remove hardcoded version strings in python code

### PR DESCRIPTION
**Pull request by Hook25**
## Description
This changes all `version` and `__version__` variables/kwargs from being hardcoded to be retrieved at runtime

## Resolved issues

Fixes: https://warthogs.atlassian.net/browse/CHECKBOX-614

## Documentation
N/A

## Tests

This was tested running metabox scenarios for both the legacy and the remote version

### CODE REVIEW

File: checkbox-ng/checkbox_ng/__init__.py [hunk 1]
```diff
@@ -24,5 +24,12 @@
...
 """
 
-__version__ = '2.6'
+from importlib_metadata import version, PackageNotFoundError
+
+try:
+    __version__ = version("checkbox-ng")
+except PackageNotFoundError:
+    import logging
+    logging.error('Failed to retieve checkbox-ng version')
```
Comment (pieqq) [inline checkbox-ng/checkbox_ng/__init__.py:33]: Typo here and below (retieve)

File: .bumpversion.cfg [hunk 2]
```diff
@@ -34,46 +34,10 @@ message = Bump version: {current_version} → {new_version}
 search = version="{current_version}"
 replace = version="{new_version}"
 
-[bumpversion:file:checkbox-ng/checkbox_ng/__init__.py]
-search = __version__ = '{current_version}'
-replace = __version__ = '{new_version}'
-
-[bumpversion:file:checkbox-ng/plainbox/__init__.py]
-search = __version__ = '{current_version}'
-replace = __version__ = '{new_version}'
```
Comment (pieqq) [inline .bumpversion.cfg:43]: If this is removed, then the current release process tools will leave `__version__` in `checkbox-ng/plainbox/__init__.py` as "2.6" forever...

**Review by pieqq — DISMISSED** [review]

File: .bumpversion.cfg (hunk 2 above)
Comment (Hook25) [reply .bumpversion.cfg:43]: `checkbox-ng/plainbox/__init__.py` should get the version number at runtime now so this line should be safe to delete

**Review by yphus — CHANGES_REQUESTED** [review]
I'd rather use `checkbox_ng.__version__` instead of `plainbox.__version__` in the providers  `manage.py`. It was really confusing when both checkbox and plainbox were used to name checkbox. Let's use the first wherever possible.

**Review by yphus — APPROVED** [review]
Thanks for the 16.04 version check. I think we're good, +1

**Pull request merged**